### PR TITLE
#445 - Extracted a new class and modified another to make it reusable.

### PR DESCRIPTION
**Pull request by vzurauskas**
PR for https://github.com/yegor256/jpeek/issues/445

Extracted a new class. Also noticed a possibility to modify another so it could be reused. That solved the issue of a disabled test which I re-enabled and removed the corresponding `todo`.

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#463](https://github.com/yegor256/jpeek/issues/463) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/C7JGJ00DP/3d27a0ac-45e3-42cf-9e4f-000a13dd26b3); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by Vatavuk** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/yegor256/jpeek/issues/463#issuecomment-619445594))

@Vatavuk The user @vatavuk[/z](https://www.0crat.com/u/vatavuk) resigned from [#463](https://github.com/yegor256/jpeek/issues/463), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
This pull request [#463](https://github.com/yegor256/jpeek/issues/463) is assigned to @fanifieiev[/z](https://www.0crat.com/u/fanifieiev), here is [why](https://www.0crat.com/footprint/C7JGJ00DP/f20c644a-8545-49ea-9bc5-2125f31fca73); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by fanifieiev** [comment]
@vzurauskas ping

**Comment by 0crat** [comment]
@fanifieiev[/z](https://www.0crat.com/u/fanifieiev) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

File: src/main/java/org/jpeek/graph/XmlMethodSignature.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+import com.jcabi.xml.XML;
+import org.cactoos.Text;
+import org.cactoos.text.Joined;
+
+/**
+ * Text signature of a class method, extracted from XML Skeleton.
+ * @since 0.30.9
+ */
+public final class XmlMethodSignature implements Text {
```
[the remark this answers is not in the document]
Comment (vzurauskas) [reply src/main/java/org/jpeek/graph/XmlMethodSignature.java]: @fanifieiev Yes, looks better now.

File: src/main/java/org/jpeek/graph/XmlMethodSignature.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+     * @param clazz The class element of XML skeleton.
+     * @param method The method element of XML skeleton.
+     */
+    public XmlMethodSignature(final XML clazz, final XML method) {
+        this.clazz = clazz;
+        this.method = method;
+    }
+
+    @Override
+    public String asString() throws Exception {
```
[the remark this answers is not in the document]
Comment (vzurauskas) [reply src/main/java/org/jpeek/graph/XmlMethodSignature.java]: @fanifieiev It now extends `TextEnvelope`, so no need for caching anymore.

**Comment by vzurauskas** [comment]
@fanifieiev Done.

File: src/main/java/org/jpeek/graph/XmlGraph.java [hunk 3]
```diff
@@ -73,28 +72,31 @@ public XmlGraph(final Skeleton skeleton) throws IOException {
      * @param skeleton XML representation on whiwh to build the graph
      * @return List of nodes
      * @throws IOException If fails
+     * @todo #445:30min This method works only for skeletons with a single
+     *  class because it assigns all methods in the skeleton to the first class
+     *  it finds. Instead, it should iterate through classes and then through
+     *  methods of each class.
      */
     private static List<Node> build(final Skeleton skeleton) throws IOException {
```
Comment (fanifieiev) [inline src/main/java/org/jpeek/graph/XmlGraph.java:80]: @vzurauskas I think this static method is a good candidate to become a class as stated here https://www.yegor256.com/2017/02/07/private-method-is-new-class.html.
So could you please leave a todo to make it a class?

Comment (vzurauskas) [reply src/main/java/org/jpeek/graph/XmlGraph.java:82]: @fanifieiev Done.

**Comment by paulodamaso** [comment]
@rultor merge

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @fanifieiev[/z](https://www.0crat.com/u/fanifieiev), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@0crat quality good

**Comment by sereshqua** [comment]
@0crat quality good

**Comment by sereshqua** [comment]
@0crat quality good

**Pull request merged**